src/pykeen/metrics/ranking.py

```diff
@@ -555,6 +567,21 @@ def __call__(self, ranks: np.ndarray, num_candidates: Optional[np.ndarray] = Non
         return np.reciprocal(stats.gmean(ranks)).item()
 
 
+def _log_cumsum_exp(a: np.ndarray):
```

Comment:
sad, torch has this https://pytorch.org/docs/stable/generated/torch.logcumsumexp.html